torch/csrc/jit/codegen/cuda/scheduler.cpp

```diff
@@ -237,6 +238,334 @@ bool scheduleFusion(Fusion* fusion, const at::ArrayRef<c10::IValue> inputs) {
+        red_tv_rf->computeAt(red_tv, 1);
+
+        red_tv->axis(0)->parallelize(ParallelType::BIDx);
+
+        // Cross-block reduction binding
+        red_tv_rf->axis(-3)->parallelize(ParallelType::BIDy);
+        red_tv->axis(-3)->parallelize(ParallelType::BIDy);
+
+        red_tv_rf->axis(-1)->parallelize(ParallelType::TIDx);
+        red_tv->axis(-1)->parallelize(ParallelType::TIDx);
+
+        red_tv_rf->axis(-2)->parallelize(ParallelType::TIDy);
+        red_tv->axis(-2)->parallelize(ParallelType::TIDy);
+
+      } else {
+        red_tv->split(-1, rparams.block_dim_x_);
+        // Split up rFactor to reduce across warps
+        red_tv->split(-2, rparams.block_dim_y_);
+
+        auto red_tv_rf = red_tv->rFactor({-3});
+        red_tv_rf->computeAt(red_tv, 1);
+
+        red_tv->axis(0)->parallelize(ParallelType::BIDx);
+
+        red_tv_rf->axis(-1)->parallelize(ParallelType::TIDx);
+        red_tv->axis(-1)->parallelize(ParallelType::TIDx);
+
+        red_tv_rf->axis(-2)->parallelize(ParallelType::TIDy);
+        red_tv->axis(-2)->parallelize(ParallelType::TIDy);
+      }
+    }
+  } else {
+    if (rparams.block_dim_y_ > 1) {
+      red_tv->split(-1, rparams.block_dim_x_);
+      if (rparams.grid_dim_y_ > 1)
+        red_tv->split(0, rparams.grid_dim_y_);
+      red_tv->split(0, rparams.block_dim_y_);
+      auto red_tv_rf = red_tv->rFactor({0});
+      red_tv_rf->axis(-1)->parallelize(ParallelType::TIDx);
+      red_tv_rf->axis(-2)->parallelize(ParallelType::BIDx);
+      if (rparams.grid_dim_y_ > 1) {
+        red_tv_rf->axis(-3)->parallelize(ParallelType::BIDy);
+        red_tv_rf->axis(-4)->parallelize(ParallelType::TIDy);
+      } else {
+        red_tv_rf->axis(-3)->parallelize(ParallelType::TIDy);
+      }
+      red_tv->axis(-1)->parallelize(ParallelType::TIDx);
+      red_tv->axis(-2)->parallelize(ParallelType::BIDx);
+      red_tv->axis(-3)->parallelize(ParallelType::TIDy);
+      if (rparams.grid_dim_y_ > 1) {
+        red_tv->axis(-3)->parallelize(ParallelType::BIDy);
+        red_tv->axis(-4)->parallelize(ParallelType::TIDy);
+      } else {
+        red_tv->axis(-3)->parallelize(ParallelType::TIDy);
+      }
+    } else {
+      red_tv->split(-1, rparams.block_dim_x_);
+      red_tv->axis(-1)->parallelize(ParallelType::TIDx);
+      red_tv->axis(-2)->parallelize(ParallelType::BIDx);
+    }
+  }
+
+  // Communicate Blocking for Kernel Launch
+  fusion->setLaunchConfig(
+      LaunchConfigType::TIDx, new Int(rparams.block_dim_x_));
+  fusion->setLaunchConfig(
+      LaunchConfigType::TIDy, new Int(rparams.block_dim_y_));
+  fusion->setLaunchConfig(LaunchConfigType::TIDz, new Int(1));
+  fusion->setLaunchConfig(LaunchConfigType::BIDx, new Int(rparams.grid_dim_x_));
+  fusion->setLaunchConfig(LaunchConfigType::BIDy, new Int(rparams.grid_dim_y_));
+  fusion->setLaunchConfig(LaunchConfigType::BIDz, new Int(1));
+  fusion->setLaunchConfig(LaunchConfigType::SharedMemory, new Int(0));
```

Comment:
We forgot to set the compatible flag here ```fusion->setLaunchConfig(LaunchConfigType::Compatible, new Int(1));```
This will result in kernel selection skipping this for any inputs. (future-ish talk)